state/executor.go

```diff
@@ -393,6 +393,12 @@ func (t *Transition) subGasLimitPrice(msg *types.Transaction) error {
 	upfrontGasCost := new(big.Int).Set(msg.GasPrice)
 	upfrontGasCost.Mul(upfrontGasCost, new(big.Int).SetUint64(msg.Gas))
 
+	// Apply EIP-1559 tx cost calculation logic
+	if msg.GasFeeCap.BitLen() > 0 {
+		upfrontGasCost = new(big.Int).SetUint64(msg.Gas)
+		upfrontGasCost = upfrontGasCost.Mul(upfrontGasCost, msg.GasFeeCap)
+	}
```

Comment:
Since we are rewriting the `uprfrontGasCost` here, you can just put in `else` the calculation of the same variable when it is not `EIP-1559` transaction.

Reply:
Agree, done